src/Misc/UnitsModM.jl

```diff
@@ -667,3 +667,36 @@ end
 
 unit_group(A::Nemo.ZZModRing) = UnitGroup(A)
 unit_group(A::Nemo.zzModRing) = UnitGroup(A)
+
+function primitive_root(n::ZZRingElem)
+  @req n >= 1 "Argument ($n) must be positive"
+  if n == 1
+    return zero(n)
+  end
+  if n == 2
+    return one(n)
+  end
+  e, _n = remove(n, 2)
+  @req n == 1 || n == 2 || n == 4 ||
+    e in [0, 1] && is_prime_power(_n) "Argument ($n) must be 2, 4, p^k, or 2p^k with p an odd prime"
```

Comment:
The code and the tests support also `n = 1`.
(Note that `Hecke.is_prime_power(1)` is `false`.)

Reply:
`1 = p^0`, so this should be included?